frontend/src/pages/interview/[interviewID].js

```diff
@@ -0,0 +1,135 @@
+/* route to page is /interview/[interviewID] where [interviewID] is a string value of an interview's id */
+import { Component } from 'react'
+import Router from 'next/router'
+import Link from 'next/link'
+import { Container, Button, Row, Col } from 'reactstrap'
+import { getInterviewByID } from '../../utils/api'
+import { ErrorMessage } from '../../components/common'
+import InterviewDetails from '../../components/interview/interviewDetails'
+import VerificationModal from '../../components/verificationModal'
+import Nav from '../../components/nav'
+import Head from '../../components/head'
+import { getKey, deleteInterview } from '../../utils/api'
+
+class InterviewDetailPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      errorMessage: '',
+      interview: null,
+      candidateID: '',
+      candidateName: '',
+      verificationModalOpen: false,
+      loading: false
+    }
+  }
+
+  async componentDidMount() {
+    const query = Router.query
+    const { result, success, message } = await getInterviewByID(query.interviewID)
+    if (!success || !result) {
+      this.setState({
+        errorMessage: message
+      })
+      return
+    }
+    this.setState({
+      interview: result.interview,
+      candidateID: result.candidateID,
+      candidateName: result.candidateName
+    })
+  }
+
+  goBack = () => {
+    Router.back()
+  }
+
+  toggleOff = () => {
+    this.setState({
+      verificationModalOpen: false
+    })
+  }
+  handleDeleteClick = () => {
+    this.setState({
+      verificationModalOpen: true
+    })
+  }
+
+  delete = async () => {
+    this.setState({
+      loading: true
+    })
+    const interview = this.state.interview
+    await deleteInterview(interview.candidate_id, interview._id)
+    this.setState({
+      verificationModalOpen: false,
+      loading: false
+    })
+    Router.push('/interviewportal')
+  }
+
+  render() {
+    const { interview, candidateName, candidateID } = this.state
+    if (this.state.errorMessage !== '' || !interview) {
```

Comment:
Use conditional rendering instead of two return statements

Reply:
Wouldn't that be a really ugly ternary (even if the return values were abstracted into their own components)? For readability purposes, if we keep it as is it's perfectly fine. No need to use syntactic sugar if it makes the code harder to follow

Reply:
@n3a9 Unless the logic is completely not understandable or very hard to understand, I would say those comments are preferences, and mention it in your comment. And in those cases, especially the ones where you can enforce them using linters/formatters I would generally not mention those unless the entire repository follows the same guidelines (say the && vs null ternary)- or if I do, then I would literally prefix the comment `nit:` or something like that because it's a preference, not something that should be enforced. We all think and process differently and maybe these 2 return statements were clearer to me than you, and it would be different for someone else reading the code. 

If you do want certain preferences to be enforced (which is awesome and I think we should be more opinionated to have a standard way of writing or code all across of h4i), then you can bring it up and we can have a discussion - and if we agree to enforce it, we'll add the rule to the linter; I'd probably let you add most rules if you'd like, since it wouldn't spend much effort into those types of preferential debates.